systest/ak_test.go

```diff
@@ -89,12 +91,13 @@ func setUpTest(t *testing.T) string {
 	// Start the AK server, but in a goroutine rather than as a separate
 	// subprocess: to support breakpoint debugging, and measure test coverage.
 	ctx, cancel := context.WithCancel(context.Background())
-	go startAKServer(ctx)
-	t.Cleanup(cancel) // Stop the AK server's goroutine.
+	go startAKServer(ctx, t)
+
+	defer cmd.App.Stop(context.Background())
```

Comment:
You can call `App.Stop()` here because of `App` is inappropriately exposed in `cmd/up.go` - see my comment about that in that file.

Also, this call breaks goroutine boundaries - it's called from one goroutine on an object in a different goroutine, and may cause race conditions.

Moreover, I don't think `defer` is used correctly here: it's guaranteed to run when the calling scope is closed, i.e. when `setUpTest()` is done, before the test begins, not after it ends.

Also, you're not checking if `App.Stop()` returns an errors, so you're not handling the condition where it failed to actually stop the AK server.